basic/airdrop-with-withdraw/src/lib.rs

```diff
@@ -0,0 +1,64 @@
+use scrypto::prelude::*;
+
+
+blueprint! {
+
+    struct AirdropWithWithdraw {
+        admin_badge: Address,
+        recipients: HashMap<Address, Vault>,
+        recipient_badge_def : ResourceDef,
+        minter_badge_vault : Vault
+    }
+
+    impl AirdropWithWithdraw {
+        
+        pub fn new() -> (Component, Bucket) {
+
+            let admin_badge = ResourceBuilder::new_fungible(DIVISIBILITY_NONE)
+                                .initial_supply_fungible(1);
+
+            let minter_badge = ResourceBuilder::new_fungible(DIVISIBILITY_NONE)
+                                               .metadata("name", "minter badge")
+                                               .initial_supply_fungible(1); 
+
+            let recipient_badge_def = ResourceBuilder::new_fungible(DIVISIBILITY_NONE)
+                                .flags(MINTABLE | INDIVIDUAL_METADATA_MUTABLE)
+                                .badge(minter_badge.resource_address(), MAY_MINT | MAY_TRANSFER | MAY_CHANGE_INDIVIDUAL_METADATA)
```

Comment:
Here you don't need the `MAY_TRANSFER` and `MAY_CHANGE_INDIVIDUAL_METADATA` authorizations. 

`MAY_TRANSFER` is only used when the flag `RESTRICTED_TRANSFER` is set.
`MAY_CHANGE_INDIVIDUAL_METADATA` and `INDIVIDUAL_METADATA_MUTABLE` are used for NFTs only, when you need to call `ResourceDef.update_nft_data()`.

Take a look at this page for more info on flags and authorization: https://docs.radixdlt.com/main/scrypto/scrypto-lang/resources.html

Reply:
done